app/app/crud/crud_user.py

```diff
@@ -48,6 +48,22 @@ def update(self, db: Session, *, db_obj: User, obj_in: Union[UserUpdate, Dict[st
             hashed_password = get_password_hash(password)
             del update_data["password"]
             update_data["hashed_password"] = hashed_password
+
+        if type_ := update_data.get('type'):
+            if db_obj.type in ('admin', 'superuser') and type_ not in ('admin', 'superuser'):
+                admin.remove(db, id=db_obj.id)
+            elif type_ == 'admin' and db_obj.type != 'admin':
+                if db_obj.type == 'superuser' and (admin_obj := admin.get(db, db_obj.id)):
+                    admin.update(db, db_obj=admin_obj, obj_in=AdminUpdate(user_id=db_obj.id, permissions=0))
+                else:
```

Comment:
Why do we need to update anything if current type and updated type are both admin?

Reply:
What? This is for other users being promoted to admins

Reply:
ah, didn't consider that usecase